src/ui_utils.c

```diff
@@ -2440,20 +2441,32 @@ void ui_init_builder(void)
 	gtk_builder_set_translation_domain(builder, GETTEXT_PACKAGE);
 
 	error = NULL;
-	interface_file = g_build_filename(app->datadir, "geany.glade", NULL);
-	if (! gtk_builder_add_from_file(builder, interface_file, &error))
+	ui_data = g_resource_lookup_data(geany_get_resource(),
+		"/org/geany/Geany/geany.glade", G_RESOURCE_LOOKUP_FLAGS_NONE, &error);
```

Comment:
Too bad we can't use [`gtk_builder_add_from_resource()`](https://developer.gnome.org/gtk3/stable/GtkBuilder.html#gtk-builder-add-from-resource)… maybe we should simply add compatibility for it on GTK < 3.4?

```c
#if !GTK_CHECK_VERSION(3, 4)
guint gtkcompat_builder_add_from_resource(GtkBuilder *builder, const gchar resource_path, GError **error)
{
    guint success = 0;
    GBytes *bytes = g_resources_lookup_data(resource_path, G_RESOURCE_LOOKUP_FLAGS_NONE, error);
    gsize bytes_size;

    if (bytes)
        success = gtk_builder_add_from_string(builder, g_bytes_get_data(bytes, &bytes_size), bytes_size, error);

    return success;
}
#define gtk_builder_add_from_resource gtkcompat_builder_add_from_resource
#endif
```

Reply:
> Too bad we can't use gtk_builder_add_from_resource()… maybe we should simply add compatibility for it on GTK < 3.4?

Yeah. Alternatively we could just add a comment there saying to change the code when the needed version is minimum supported.

I think this specific (example) code is missing an `#else`, and also I think the `gtk_build_add_from_string` call is not OK because of the undefined ordering of parameter evaluation in C (which is why I did it with two calls in the patch).

Reply:
> Yeah. Alternatively we could just add a comment there saying to change the code when the needed version is minimum supported.

What I like with the gtkcompat thing is that we just have to drop the compat code when it's not needed anymore, we don't need to touch caller code.

> I think this specific (example) code is missing an `#else`

It's not, as it's replacing the GTK API with the `#define`, so the else if just the normal GTK 3.4 call.

> `gtk_build_add_from_string` call is not OK because of the undefined ordering of parameter evaluation in C (which is why I did it with two calls in the patch).

Good point

Reply:
> What I like with the gtkcompat thing is that we just have to drop the compat code when it's not needed anymore

Seems OK to me. I'll try and get to this.

> It's not, ...

Ah, I didn't notice the differing prefixes on the names.

Reply:
I'm on it, give me a sec and you won't have to bother ;)

Reply:
@codebrainz https://github.com/b4n/geany/tree/codebrainz/gresource